src/app/rundown/components/piece/piece.component.ts

```diff
@@ -11,5 +11,76 @@ export class PieceComponent {
   @Input()
   public piece: Piece
 
-  constructor() { }
+  @Input()
+  public pixelsPerSecond: number
+
+  @Input()
+  public partDuration: number
+
+  @Input()
+  public expectedPartDuration: number
+
+  public getPieceWidthInPixels(): string {
+    const width = this.getPieceWidth()
+    return `${width}px`
+  }
+
+  private getPieceWidth(): number {
+    const duration  = this.piece.duration ?? (this.partDuration - this.piece.start)
+    return Math.floor(this.pixelsPerSecond * duration / 1000)
+  }
+
+  public getOffsetInPixels(): string {
+    const offset = this.getOffset()
+    return `${offset}px`
+  }
+
+  private getOffset(): number {
+    return Math.floor(this.pixelsPerSecond * (this.piece.start ?? 0) / 1000)
+  }
+
+  public getPieceTypeModifierClass(): string {
+    const pieceType = this.getPieceType()
+    return `piece--${pieceType}`
+  }
+
+  // TODO: Use either piece.type or source layer types for determining style.
+  public getPieceType(): string {
+    const layerType = this.piece.layer.replace(/^studio\d+_/, '')
+    switch (layerType) {
+      case 'camera':
+        return 'camera'
+      case 'script':
+        return 'manus'
+      case 'graphicsLower':
+      case 'graphicsIdent':
+      case 'graphicsHeadline':
+      case 'graphicsTop':
+        return 'lower-third'
+      case 'graphicsTelefon':
+      case 'selected_graphicsFull':
+      case 'graphicsTema':
+      case 'overlay':
+      case 'pilot':
+      case 'pilotOverlay':
+      case 'wall_graphics':
+      case 'graphic_show_lifecycle':
+        return 'graphics'
+      case 'live':
+        return 'remote'
+      case 'clip':
+      case 'selected-clip':
+        return 'clip'
+      case 'voiceover':
+      case 'selected_voiceover':
+        return 'live-speak'
+      case 'local':
+        return 'local'
+      case 'dve':
+      case 'dve_adlib':
+        return 'dve'
+      default:
+        return 'unknown'
+    }
```

Comment:
This needs to be something else than magic strings.
Also, I don't think sourceLayers should be used for this.

Reply:
Before the task is set to done, we will have addressed this. But it requires that piece.type is populated in the backend.